programs/clearing_house/src/lib.rs

```diff
@@ -544,13 +552,25 @@ pub mod clearing_house {
                     amount
                 };
 
-            controller::bank_balance::update_bank_balances(
+            // prevents withdraw when limits hit
+            controller::bank_balance::update_bank_balances_with_limits(
                 amount as u128,
                 &BankBalanceType::Borrow,
                 bank,
                 user_bank_balance,
             )?;
 
+            // prevents borrow when bank market's oracle invalid
+            if bank.has_market {
```

Comment:
can do `if let Some(perp_market_index) = bank.perp_market_index` if we make it an Option (see comment on bank.rs

Reply:
I removed update_amms in this function, so I think we need to add back the logic to update the amm for the banks perp market? (can help with that)

Reply:
i dont see where you update the amm before this?